main.tf

```diff
@@ -1,6 +1,14 @@
 data "aws_partition" "current" {}
 data "aws_caller_identity" "current" {}
 
+data "aws_iam_session_context" "current" {
```

Comment:
Should we consider using this resource instead of an existing one in other modules also? WDYT?

Reply:
yes, I think anywhere that we are using `data.aws_caller_identity.current.arn` to get the current identity's ARN, we should use this session context data source to ensure we are getting the true ARN and not the ARN of the STS session